src/datasources/config-api/config-api.service.ts

```diff
@@ -1,32 +1,62 @@
 import { Injectable } from '@nestjs/common';
-import { Page } from './entities/page.entity';
-import { Chain } from './entities/chain.entity';
 import { Inject } from '@nestjs/common';
 import { IConfigurationService } from '../../common/config/configuration.service.interface';
 import { CacheFirstDataSource } from '../cache/cache.first.data.source';
+import { Page } from '../../common/entities/page.entity';
+import { Chain } from '../../chains/entities';
+import { DefinedError, ValidateFunction } from 'ajv';
+import { ValidationErrorFactory } from '../errors/validation-error-factory';
+import { JsonSchemaService } from '../common/json-schema.service';
+import {
+  chainSchema,
+  nativeCurrencySchema,
+} from '../../chains/entities/schemas/chain.schema';
 
 @Injectable()
 export class ConfigApi {
   private readonly baseUri: string;
+  private readonly isValidChain: ValidateFunction<Chain>;
 
   constructor(
     private readonly dataSource: CacheFirstDataSource,
     @Inject(IConfigurationService)
     private readonly configurationService: IConfigurationService,
+    private readonly validationErrorFactory: ValidationErrorFactory,
+    private readonly jsonSchemaService: JsonSchemaService,
   ) {
     this.baseUri =
       this.configurationService.getOrThrow<string>('safeConfig.baseUri');
+    this.jsonSchemaService.addSchema(nativeCurrencySchema, 'nativeCurrency');
+    this.isValidChain = this.jsonSchemaService.compile(
+      chainSchema,
+    ) as ValidateFunction<Chain>;
   }
 
   async getChains(): Promise<Page<Chain>> {
     const key = 'chains'; // TODO key is not final
-    const url = this.baseUri + '/api/v1/chains';
-    return await this.dataSource.get(key, url);
+    const url = `${this.baseUri}/api/v1/chains`;
+    const page: Page<Chain> = await this.dataSource.get(key, url);
+
+    if (!page?.results.every((chain) => this.isValidChain(chain))) {
+      // TODO: probably we want to invalidate cache at this point
+      const errors = this.isValidChain.errors as DefinedError[];
+      throw this.validationErrorFactory.from(errors);
+    }
```

Comment:
We should test when a chain is not valid.

Reply:
Sorry I forgot to say this in our prior conversation but this class doesn't have any test yet. I'm adding these tests to the controllers in [1a50781](https://github.com/5afe/safe-client-gateway-nest/pull/66/commits/1a5078151be359884f206f1691f9feef39010174) 🙂